src/Helpers/JWKFetcher.php

```diff
@@ -38,43 +45,69 @@ public function __construct(CacheHandler $cache = null, $guzzleOptions = [])
     }
 
     /**
+     * Convert a certificate to PEM.
+     *
+     * @param string $cert Certificate to convert.
      *
-     * @param  string $cert
      * @return string
      */
     protected function convertCertToPem($cert)
     {
-        return '-----BEGIN CERTIFICATE-----'.PHP_EOL.chunk_split($cert, 64, PHP_EOL).'-----END CERTIFICATE-----'.PHP_EOL;
+        $output  = '-----BEGIN CERTIFICATE-----'.PHP_EOL;
+        $output .= chunk_split($cert, 64, PHP_EOL);
+        $output .= '-----END CERTIFICATE-----'.PHP_EOL;
+        return $output;
     }
 
     /**
+     * Fetch JWKS from the token issuer.
      *
-     * @param string $iss
+     * @param string      $domain    Typically the issuer for the JWKS being fetched; include a trailing slash.
+     * @param null|string $jwks_path Path to the JWKS from the issuer domain; pass null to use $domain as the URL.
      *
      * @return array|mixed|null
-     *
-     * @throws \Exception
      */
-    public function fetchKeys($iss)
+    public function fetchKeys($domain, $jwks_path = '.well-known/jwks.json')
     {
-        $url = "{$iss}.well-known/jwks.json";
+        // If the path parameter is empty, use the $domain parameter as a complete URL.
+        if (empty( $jwks_path )) {
+            $domain_parsed = parse_url( $domain );
+            $domain        = $domain_parsed['scheme'].'://'.$domain_parsed['host'];
+            $domain       .= empty( $domain_parsed['port'] ) ? '' : ':'.$domain_parsed['port'];
+            $domain       .= '/';
+            $jwks_path     = empty( $domain_parsed['path'] ) ? '' : substr( $domain_parsed['path'], 1 );
+        }
+
+        // Check if we have a cache entry for this URL.
+        $url    = $domain.$jwks_path;
+        $secret = $this->cache->get($url);
```

Comment:
I'm still not convinced that this is a good "key" to cache this. JWK may rotate / you might not have the latest version. So even if you have a cache for that URI, the JWK might be missing and you might want to attempt to download a fresh JWKS again. So knowing the `kid` you're looking for in that set is something critical IMO. 

- What about having 3 params: (domain, kid, jwks_path with default)? 
- Also how often does the cache expire?

Reply:
> I'm still not convinced that this is a good "key" to cache this.

I agree but that's the key we started with. Not a huge problem to change the key but also not critical IMHO. 

> What about having 3 params: (domain, kid, jwks_path with default)?

That was my thought when we add the `kid`. I think the domain and path should sit together, though. I also want to separate the "get key" logic from the "get JWKS data" logic. I have a backlog task for that. 

> Also how often does the cache expire?

Up to the developer, the default is no cache.

Reply:
> I think the domain and path should sit together, though

I separated them because you will normally not pass the jwks path. That's another use case, but not the "happy path" one. If you keep them together as you say and you don't use a jwks_path, you would need to pass null as second parameter to be able to call this function, right? that doesn't look good 😛 I rather have it at the end with a default value when missing, so that you can call this function with just 2 params. That's the idea behind my signature suggestion.

> I also want to separate the "get key" logic from the "get JWKS data" logic.

Fine by me. I don't think you can change this now but, obtaining the "JWKS data" is not something you might want to expose publicly. Just the "get key by id"